Mention parameter type for customizing template (#2155)
theforeman/foreman-documentation: guides/common/modules/ref_customizing-the-registration-templates.adoc

ifndef::satellite[]
* The `host_registration_insights` parameter is used in the `insights` snippet, the default value is `false`.
Set the parameter type to `boolean` to override the parameter value.
endif::[]
ifdef::satellite[]
* The `host_registration_insights` parameter is used in the `insights` snippet, the default value is `true`.
Set the parameter type to `boolean` to override the parameter value.
endif::[]

> If the parameter is set to `false`, it prevents hosts from appearing in the Red{nbsp}Hat Inventory reports and also the Insights client from uploading reports to the Inventory of your {RHCloud}.

Reviewer: Should this say "Red Hat Insights" instead of "Red Hat Inventory reports"? @mjivraja

> If the parameter is set to `false`, it prevents hosts from appearing in the Red{nbsp}Hat Inventory reports and also the Insights client from uploading reports to the Inventory of your {RHCloud}.

Reviewer: Because when I look into the WebUI, there are the following items under *Configure* > *RH Cloud*: *Inventory Upload* and *Insights*.

> If the parameter is set to `false`, it prevents hosts from appearing in the Red{nbsp}Hat Inventory reports and also the Insights client from uploading reports to the Inventory of your {RHCloud}.

Reviewer: ```suggestion
If the parameter is set to `false`, it prevents hosts from appearing in the Red{nbsp}Hat Insights and also the Insights client from uploading Inventory reports to your {RHCloud}.
```
Maybe like this? Pls, check this with SMEs.

> If the parameter is set to `false`, it prevents hosts from appearing in the Red{nbsp}Hat Inventory reports and also the Insights client from uploading reports to the Inventory of your {RHCloud}.

Reviewer 2: @ShimShtein perhaps you can enlighten us on this.

> If the parameter is set to `false`, it prevents hosts from appearing in the Red{nbsp}Hat Inventory reports and also the Insights client from uploading reports to the Inventory of your {RHCloud}.

Author: I guess, hosts are still part of RH Insights despite the parameter being false. But definitely, let's wait for @ShimShtein

> If the parameter is set to `false`, it prevents hosts from appearing in the Red{nbsp}Hat Inventory reports and also the Insights client from uploading reports to the Inventory of your {RHCloud}.

Reviewer 3: The inventory on the cloud side accepts many types of reports. As long as at least one reporter continues to report about a host, it will appear in the cloud inventory.
Here we are talking about two reporters that are controlled by Satellite: the rh_cloud (Satellite) generated reports and `insights-client` reports. While those are the main contributors to the inventory, they are not the only ones that exist, and that's why I am a bit reluctant about documenting this parameter as something that will prevent the entry in the inventory for sure. We want to indicate that this parameter changes the behavior of those two reports.

> If the parameter is set to `false`, it prevents hosts from appearing in the Red{nbsp}Hat Inventory reports and also the Insights client from uploading reports to the Inventory of your {RHCloud}.

Reviewer 3: Maybe something along the lines of
```
It prevents Satellite and the Insights client from uploading Inventory reports to your {RHCloud}.
```
Does this make sense?

ifndef::satellite[]
* The `host_registration_insights` parameter is used in the `insights` snippet, the default value is `false`.

> The type must be set to `Boolean` to override the parameter value.

Reviewer: ```suggestion
Set the parameter type to `boolean` to override the parameter value.
```

endif::[]
ifdef::satellite[]
* The `host_registration_insights` parameter is used in the `insights` snippet, the default value is `true`.

> The type must be set to `Boolean` to override the parameter value.

Reviewer: ```suggestion
Set the parameter type to `boolean` to override the parameter value.
```

> If the parameter is set to `false`, it prevents hosts from appearing in the `rh-cloud reports` and also restricts `insights-client` from uploading its reports.

Reviewer: ```suggestion
If the parameter is set to `false`, it prevents hosts from appearing in the Red Hat Cloud Console reports and also the `insights-client` from uploading reports to {Project}.
```
I'm not sure, about Insights client uploading reports to {Project}, verify with a SME.
Also, this sentence is quite important and I think it should be included for both non-Satellite and Satellite alike.

Reviewer 3: I can confirm. The code looks for this parameter and will reject `insights-client` reports if the parameter is set to `false`.

Reviewer 3: As for `Red Hat Cloud Console reports`: I think it's a bit vague. There are two types of reports: one generated by insights-client on the host itself, and the other generated by the rh_cloud plugin. Both are contributing to the Red Hat Inventory. While `insights-client` reports are used to generate recommendations, rh_cloud reports are used for inventory and RHEL usage reporting.

Reviewer 4: The word "restricted" is a bit confusing for few reason. Because the parameter influence the system during its deployment. A user can change the insights-client whenever they want after that. Setting the parameter to "false" just do not enable insights and therefore nothing is sent. But it does not restrict per say. 

insights-client do no send data to Satellite in a way to be consumed by it when the parameter is set to true, it uses Satellite as a proxy. 

This parameter is not something that influence Satellite itself, it influences what kickstart scripts are given to the client system getting deployed via Satellite.

Reviewer 3: @cmarinea not only during the deployment. I have specifically added code that would prevent Foreman from forwarding reports from `insights-client` to the cloud: https://github.com/theforeman/foreman_rh_cloud/blob/foreman_3_7/app/controllers/insights_cloud/api/machine_telemetries_controller.rb#L76

Of course you are correct about its influence during the deployment, but it's additional functionality that we have added specifically to the rh_cloud plugin.

Reviewer 4: @ShimShtein So just to confirm the parameter intended to influence the kickstart template file "host_registration_insights" has signification elsewhere in the code?

If a user enable insights-client later on, the Satellite will not act as a proxy?

For the rh_cloud plugin, I thought it was another configuration completely.  I may be a little confused :)

Reviewer 3: > So just to confirm the parameter intended to influence the kickstart template file "host_registration_insights" has signification elsewhere in the code?

The parameter intended to regulate a feature: insights-client reports. This means that it will influence everything insights-client related, be it provisioning or reporting. Imagine a user who is using RHV templates to provision new machines (no provisioning is done through the kickstart process) , if the user sets the parameter to false, it would be rational that insights-client will not try to report to the cloud, even if it's part of the template.

> If a user enable insights-client later on, the Satellite will not act as a proxy?

As far as I know insights-client has a way to discover it consumes content from Satellite, and if it discovers this fact, it will use Satellite as a proxy. I think there is a way to prevent this automatic discovery process and point it directly to the cloud, but it is definitely not the default configuration.

> For the rh_cloud plugin, I thought it was another configuration completely. I may be a little confused :)

I have tried to reduce the amount of knobs and levers in rh_cloud, and if I identified an already existing way of expressing user's idea, I used it in the rh_cloud plugin. There are a couple of configuration options specific to rh_cloud, but those are there only because rh_cloud introduced new features that required its own configuration.

Author: Pushed. I hope we're closer if not good to go. :crossed_fingers:

> If the parameter is set to `false`, it prevents hosts from appearing in the {Team} Inventory reports and also the `insights-client` from uploading reports to `rh-cloud`.

Reviewer: ```suggestion
If the parameter is set to `false`, it prevents hosts from appearing in the {Team} Inventory reports and also the Insights client from uploading reports to RH Cloud.
```
I'm suggesting a couple of formatting improvements. "RH Cloud" is found in the WebUI, so I think it's vital to spell it like this.
Again, I suggest taking this line out of the condition, because it's relevant for all build targets.

Author: @Lennonka We are discussing [here](https://github.com/theforeman/foreman-documentation/pull/2180#discussion_r1187306218) the potential possibility of adding attributes for RH Cloud. Should we wait for it or we shall move forward here?

Reviewer: @mjivraja Perhaps we should introduce a new attribute for "Red Hat Cloud" as well in another PR so that we start using the product name consistently? #2180 is about something different if I understand correctly.

Author: In this [PR](https://github.com/theforeman/foreman-documentation/pull/2180#pullrequestreview-1417989076) Maximilian shares sensible data to avoid introducing the new attribute for RH Cloud. 

However, I also agree with your purpose to introduce the attribute for "Red Hat Cloud" to start using the product name consistently.

Reviewer: @mjivraja Yes, but he's talking about an attribute for all builds and for a program name in the CLI. We can add an attribute for the product title only for the Satellite build.

> If the parameter is set to `false`, it prevents hosts from appearing in the {Team} Inventory reports and also the Insights client from uploading reports to the Inventory of your {RHCloud}.

Reviewer: This line applies to all builds.

Reviewer: And I think this is the case where you want to use `Red Hat` instead of `{Team}` :)

Author: Indeed :smiley:
